manta-trusted-setup/src/groth16/ceremony/serde.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019-2022 Manta Network.
+// This file is part of manta-rs.
+//
+// manta-rs is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// manta-rs is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with manta-rs.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Serde
+
+use manta_crypto::arkworks::serialize::{CanonicalDeserialize, CanonicalSerialize};
+use manta_util::{
+    serde::{Deserialize, Deserializer, Serialize, Serializer},
+    BoxArray,
+};
+
+/// Uses `serializer` to serialize `data` that implements `CanonicalSerialize`.
+#[inline]
+pub fn serialize_element<T, S>(data: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    T: CanonicalSerialize,
+    S: Serializer,
+{
+    let mut bytes = Vec::new();
+    data.serialize(&mut bytes).unwrap();
```

Comment:
Use S::Error

Reply:
move to manta-crypto.
Replace [this line](https://github.com/Manta-Network/manta-rs/blob/main/manta-crypto/src/arkworks/mod.rs#L22) with `pub mod serialize` and add these two functions